merlin/models/tf/core.py

```diff
@@ -1788,14 +1795,22 @@ def __init__(
         self.task_block = task_block
         self._task_name = task_name
         self.pre = pre
-        self.pre_metrics = pre_metrics
+        self.pre_eval_topk = pre_eval_topk
+
+        if metrics is not None:
+            if isinstance(metrics, SequenceCollection):
+                metrics = list(metrics)
+            else:
+                metrics = [metrics]
 
         create_metrics = self._create_metrics
         self.eval_metrics = create_metrics(metrics) if metrics else []
         self.prediction_metrics = create_metrics(prediction_metrics) if prediction_metrics else []
         self.label_metrics = create_metrics(label_metrics) if label_metrics else []
         self.loss_metrics = create_metrics(loss_metrics) if loss_metrics else []
-        self.compute_train_metrics = compute_train_metrics
+
+    def set_pre_eval_topk(self, block):
```

Comment:
Why do we want a setter for this? If it’s for safety we might want to make the variable private and expose it as a property.

Reply:
Yes, it is to protect the internal variable. I did like you suggested